File: brute.py
with open("random_output.txt") as f:
    numbers = []
    for _ in range(100):
        numbers.append(float(f.readline()))

from WichmanHill import r8_random as rand

print(rand(100, 200, 300))


# s1 = s2 = s3 = 100
# for i in range(5):
#     val, s1, s2, s3 = rand(s1, s2, s3)
#     print(val)

first=0.44960042304882464
0.44960041783776544
0.44960042394708233
val=1

for i in range(100, 1000):
    for j in range(100, 1000):
        for k in range(100, 1000):
            tmp,a,b,c=rand(i,j,k)
            if abs(tmp-first) < abs(val-first):
                val=tmp
                print("changing to %s %s %s (val=%s)" % (i,j,k,val))


# s1, s2, s3 = 533, 888, 963
# for i in range(100):
#     val, s1, s2, s3 = rand(s1, s2, s3)
#     print(val)

# for a in range(100, 1000):
#     for b in range(100, 1000):
#         for c in range(100, 1000):
#             # print("new:", a, b, c)
#             s1, s2, s3 = a, b, c
#             for i in range(100):
#                 val, s1, s2, s3 = rand(s1, s2, s3)
#                 # if round(val*10000000)//10000000 != round(numbers[i]*10000000)//10000000:
#                 if val != numbers[i]:
#                     break
#             else:
#                 print(a, b, c)
#                 break
#         else: continue
#         break
#     else: continue
#     break

File: brute.cpp
#include <iostream>
#include <tuple>


using namespace std;

tuple<double, int, int, int> rand(int s1, int s2, int s3) {
    s1 = ( ( 171 * s1 ) % 30269 );
    s2 = ( ( 172 * s2 ) % 30307 );
    s3 = ( ( 170 * s3 ) % 30323 );

    double value = float ( s1 ) / 30269.0 + float ( s2 ) / 30307.0 + float ( s3 ) / 30323.0;

    // value = ( value % 1);

    return tuple<double, int, int, int>{value, s1, s2, s3};
}

int main() {

}